nevergrad/functions/images/imagelosses.py

```diff
@@ -1,35 +1,102 @@
+import cv2
+import imquality.brisque as brisque
+import lpips
 import os
+import torch
 import typing as tp
 import numpy as np
 from nevergrad.functions.base import UnsupportedExperiment
+from nevergrad.common.decorators import Registry
+
+
+registry: Registry[tp.Any] = Registry()
 
 
 class ImageLoss:
     def __init__(self, reference: tp.Optional[np.ndarray] = None) -> None:
         pass
 
     def __call__(self, img: np.ndarray) -> float:
-        raise NotImplementedError
+        raise NotImplementedError(f"__call__ undefined in class {type(self)}")
 
 
-class SumAbsoluteDifferences(ImageLoss):
+class ImageLossWithReference(ImageLoss):
     def __init__(self, reference: np.ndarray) -> None:
         if reference is None:
             raise ValueError("A reference is required")
         self.reference = reference
         super().__init__(reference)
+        assert len(self.reference.shape) == 3, self.reference.shape
```

Comment:
If you want to make sure the format of the reference is the same for every loss, maybe add something about the values. 
```suggestion
        assert len(self.reference.shape) == 3, self.reference.shape
        assert len(self.reference.min()) >= 0 and len(self.reference.max()) <= 255, f"The reference should contain values between 0 and 255, they were between {self.reference.min()} and {self.reference.max()}"
```

Reply:
I guess the "len" here is a typo, but yes that makes sense.

Reply:
Done.